packages/popover/src/vaadin-popover.js

```diff
@@ -131,10 +155,19 @@ class Popover extends PopoverPositionMixin(
     this._overlayElement = this.shadowRoot.querySelector('vaadin-popover-overlay');
   }
 
+  /** @protected */
+  connectedCallback() {
+    super.connectedCallback();
+
+    document.addEventListener('click', this.__onGlobalClick);
+  }
+
   /** @protected */
   disconnectedCallback() {
     super.disconnectedCallback();
 
+    document.removeEventListener('click', this.__onGlobalClick);
```

Comment:
nit: untested

Reply:
Added a test to check that `click` listener is removed on detach. Not sure if there is a better way to test this.